ext/js/language/dictionary-importer.js

```diff
@@ -113,11 +113,15 @@ export class DictionaryImporter {
         const dataBankSchemas = this._getDataBankSchemas(version);
 
         // Files
-        const termFiles = this._getArchiveFiles(fileMap, 'term_bank_?.json');
-        const termMetaFiles = this._getArchiveFiles(fileMap, 'term_meta_bank_?.json');
-        const kanjiFiles = this._getArchiveFiles(fileMap, 'kanji_bank_?.json');
-        const kanjiMetaFiles = this._getArchiveFiles(fileMap, 'kanji_meta_bank_?.json');
-        const tagFiles = this._getArchiveFiles(fileMap, 'tag_bank_?.json');
+        /** @type {import('dictionary-importer').QueryDetails} */
+        const queryDetails = new Map([
+            ['termFiles', 'term_bank_?.json'],
```

Comment:
I wonder if these should just be regular expressions. This is a pretty old method of detecting file name matches in the codebase, and I don't recall there being any specific reason why this was done vs just a regex match of like `/^term_bank_(\d+)\.json$/g`. This would simplify this code a bit while we're here.

Here's how it originally looked, so I don't see any reason why I explicitly kept it like that other than "because it's already like that."
https://github.com/themoeway/yomitan/blame/63a775ebcafc8d0477bc267ef39e40c4d3283666/ext/bg/js/database.js#L525C6-L529

Reply:
Maybe there is an overhead to matching regex for such a simple expression, bc it would be called for possibly thousands of files. I dont know too much about regex though.

Reply:
It is negligible even at the extremely unlikely scale of thousands of or even millions of files. There is also realistically no need to have that many files, and even further, there's really not much reason to have more than 1 file per data type anyways. It was originally set up that way to make batching database inserts a bit easier, but that can and has been resolved on the extension's side.

The entire zip file is basically loaded into RAM anyways, so there's not a hugely significant memory saving; not sure if zip.js has changed things there. Maybe the JSON deserialization could be having an effect if multiple files are used and the JS garbage collector works well (related: #381).

Reply:
Thanks. I will include the changes